pkg/jsonschema/json-schema.go

```diff
@@ -10,23 +10,39 @@ import (
 	"github.com/HewlettPackard/terraschema/pkg/reader"
 )
 
-func CreateSchema(path string, strict bool) (map[string]any, error) {
+type CreateSchemaOptions struct {
+	RequireAll                bool
+	AllowAdditionalProperties bool
+	AllowEmpty                bool
+}
+
+func CreateSchema(path string, options CreateSchemaOptions) (map[string]any, error) {
 	schemaOut := make(map[string]any)
 
 	varMap, err := reader.GetVarMap(path)
-	// GetVarMaps returns an error if no .tf files are found in the directory. We
-	// ignore this error for now.
-	if err != nil && !errors.Is(err, reader.ErrFilesNotFound) {
-		return schemaOut, fmt.Errorf("error reading tf files at %s: %w", path, err)
+	if err != nil {
+		if errors.Is(err, reader.ErrFilesNotFound) {
+			if options.AllowEmpty {
+				fmt.Printf("No tf files were found in %q, creating empty schema\n", path)
```

Comment:
nit: capitalization (No/no)

Reply:
Actually, this one will need to be changed to be more in line with a future PR. I'll add the change here.